presto-main/src/main/java/com/facebook/presto/server/PrestoServer.java

```diff
@@ -131,6 +131,24 @@ public void run()
             injector.getInstance(Announcer.class).start();
 
             log.info("======== SERVER STARTED ========");
+
+            ServerConfig serverConfig = injector.getInstance(ServerConfig.class);
+
+            if (serverConfig.isCoordinator() && serverConfig.isEnabledUIonSecondaryPort()) {
+                Bootstrap uIApp = new Bootstrap(new NodeModule(),
+                                                new DiscoveryModule(),
+                                                new JsonModule(),
+                                                new JaxrsModule(true),
+                                                new JmxModule(),
+                                                new JmxHttpModule(),
+                                                new JsonEventModule(),
+                                                new HttpEventModule(),
+                                                new ServerSecurityModule(),
+                                                new CoordinatorUIHttpServerModule(injector));
+                Injector uIInjector = uIApp.doNotInitializeLogging().initialize();
+
+                log.info("======== SECONDARY UI STARTED ========");
```

Comment:
I see what's happening now. You seem to be restarting a bunch of things twice on another port. Instead could break out the UI into its own configurable module so we could just include it in the list of Modules above and have it start as they do? Or does that not work because all the modules in the Bootstrap require a single http port?